### PR TITLE
HAI-2113 Adjust tram traffic hindrance index calculation

### PR DESCRIPTION
**Pull request by umeetiusbaar**
# Description

Changed tram traffic hindrance calculation based on new specifications: https://helsinkisolutionoffice.atlassian.net/wiki/spaces/HAI/pages/8237088854
* new tables `tormays_tram_infra_polys` and `tormays_tram_lines_polys` instead of old `tormays_trams_polys`
* new calculation rules

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [x] Other

# Instructions for testing
1. run `scripts/HEL-GIS-data-import.sh` or at least the following (on VPN):
```
docker run --rm --network=haitaton-backend_backbone osgeo/gdal:alpine-small-latest ogr2ogr -overwrite -f PostgreSQL PG:"dbname='haitaton' host='db' port='5432' user='haitaton_user' password='haitaton'" http://haitaton-fileshare-dev.internal.apps.arodevtest.hel.fi/tormays_tram_infra_polys.gpkg
docker run --rm --network=haitaton-backend_backbone osgeo/gdal:alpine-small-latest ogr2ogr -overwrite -f PostgreSQL PG:"dbname='haitaton' host='db' port='5432' user='haitaton_user' password='haitaton'" http://haitaton-fileshare-dev.internal.apps.arodevtest.hel.fi/tormays_tram_lines_polys.gpkg
```
2. for live testing Haitaton UI may be used. Tram indexes should be 0 (no tram tracks), 3 (tram tracks but no active tram lines) or 5 (active tram lines). QGIS can be used to see where the tram infra and lines are located.

# Checklist:

- [x] I have written new tests (if applicable)
- [x] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
 or other location: 

# Other relevant info
NOTICE! New gis data has to be loaded into the database before merging this!

### CODE REVIEW

**Comment by umeetiusbaar** [comment]
SonarCloud complains about too many duplicate lines but they are from SQL files intended for setting up local database in Docker or integration tests. These files contain small amount of rows for new database tables.

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/TormaystarkasteluTormaysServiceITest.kt [hunk 1]
```diff
@@ -115,8 +115,8 @@ internal class TormaystarkasteluTormaysServicePGITest : DatabaseTest() {
     @Test
     fun raitiotiet() {
         val geometriaIds = createHankeGeometria()
-        assertThat(tormaysService.maxIntersectingTramByLaneType(geometriaIds))
-            .isEqualTo(TormaystarkasteluRaitiotiekaistatyyppi.JAETTU.value)
+        assertThat(tormaysService.anyIntersectsWithTramLines(geometriaIds)).isTrue()
+        assertThat(tormaysService.anyIntersectsWithTramInfra(geometriaIds)).isTrue()
```
Comment (corvidian) [inline services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/TormaystarkasteluTormaysServiceITest.kt:119]: Is it enough to check that both happen to be true in that one spot? Should there be tests for negative cases?

File: services/hanke-service/src/main/resources/db/changelog/changesets/057-drop-tormays_trams_polys.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+databaseChangeLog:
+  - changeSet:
+      id: 017-drop-tormays_trams_polys
```
[the remark this answers is not in the document]
Comment (umeetiusbaar) [reply services/hanke-service/src/main/resources/db/changelog/changesets/057-drop-tormays_trams_polys.yml]: Fixed

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/TormaystarkasteluTormaysServiceITest.kt (hunk 1 above)
Comment (umeetiusbaar) [reply services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/TormaystarkasteluTormaysServiceITest.kt]: Added a new geometry that does not intersect with any tormays data. Now there are two test geometries, one for central location (Kaivokatu) and one for out-of-all-infra location (Mustikkamaa).

**Review by corvidian — APPROVED** [review]

**Pull request merged**